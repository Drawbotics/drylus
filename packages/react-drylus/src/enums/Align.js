import Enum from '@drawbotics/enums';


const Align = new Enum(
  'LEFT',
  'CENTER',
  'RIGHT',
);


export default Align;
