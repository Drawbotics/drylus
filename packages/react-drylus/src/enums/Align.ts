export enum Align {
  LEFT = 'LEFT',
  CENTER = 'CENTER',
  RIGHT = 'RIGHT',
  JUSTIFY = 'JUSTIFY',
}
