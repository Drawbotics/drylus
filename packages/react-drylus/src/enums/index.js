export Category from './Category';
export Tier from './Tier';
export Size from './Size';
export Position from './Position';
export Align from './Align';
export Color from './Color';
export Shade from './Shade';
