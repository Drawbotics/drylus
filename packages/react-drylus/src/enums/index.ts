export * from './Category';
export * from './Tier';
export * from './Size';
export * from './Position';
export * from './Align';
export * from './Color';
export * from './Shade';
