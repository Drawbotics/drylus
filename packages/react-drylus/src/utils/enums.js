import camelCase from 'lodash/camelCase';

import { Category, Color, Shade, Tier } from '../enums';

export function getEnumAsClass(enumVal) {
  return camelCase(enumVal?.description?.toLowerCase());
}

export function colorEnumToCategory(enumVal) {
  switch (enumVal) {
    case Color.RED:
      return Category.DANGER;
    case Color.GREEN:
      return Category.SUCCESS;
    case Color.ORANGE:
      return Category.WARNING;
    case Color.BLUE:
      return Category.INFO;
  }
  return enumVal;
}

export function categoryEnumToColor(enumVal) {
  switch (enumVal) {
    case Category.DANGER:
      return Color.RED;
    case Category.SUCCESS:
      return Color.GREEN;
    case Category.WARNING:
      return Color.ORANGE;
    case Category.INFO:
      return Color.BLUE;
  }
  return enumVal;
}

export function shadeEnumToTier(enumVal) {
  switch (enumVal) {
    case Shade.DARK:
      return Tier.PRIMARY;
    case Shade.MEDIUM:
      return Tier.SECONDARY;
    case Shade.LIGHT:
      return Tier.TERTIARY;
  }
  return enumVal;
}

export function getIconForCategory(category) {
  switch (category) {
    case Category.DANGER:
      return 'alert-circle';
    case Category.SUCCESS:
      return 'check-circle';
    case Category.WARNING:
      return 'alert-triangle';
    default:
      return 'info';
  }
}
