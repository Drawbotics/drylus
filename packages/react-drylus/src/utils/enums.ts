import camelCase from 'lodash/camelCase';

import { Category, Color, Position, Shade, Size, Tier } from '../enums';

type Enum = Category | Color | Shade | Tier | Size | Position;

export function getEnumAsClass<T extends Record<string, string>>(enumVal?: Enum): keyof T {
  return camelCase(enumVal?.toLowerCase());
}

export function colorEnumToCategory(enumVal: Color): Category | undefined {
  switch (enumVal) {
    case Color.RED:
      return Category.DANGER;
    case Color.GREEN:
      return Category.SUCCESS;
    case Color.ORANGE:
      return Category.WARNING;
    case Color.BLUE:
      return Category.INFO;
  }
}

export function categoryEnumToColor(enumVal: Category): Color | undefined {
  switch (enumVal) {
    case Category.DANGER:
      return Color.RED;
    case Category.SUCCESS:
      return Color.GREEN;
    case Category.WARNING:
      return Color.ORANGE;
    case Category.INFO:
      return Color.BLUE;
  }
}

export function shadeEnumToTier(enumVal: Shade): Tier | undefined {
  switch (enumVal) {
    case Shade.DARK:
      return Tier.PRIMARY;
    case Shade.MEDIUM:
      return Tier.SECONDARY;
    case Shade.LIGHT:
      return Tier.TERTIARY;
  }
}

export function getIconForCategory(category: Category): string {
  switch (category) {
    case Category.DANGER:
      return 'alert-circle';
    case Category.SUCCESS:
      return 'check-circle';
    case Category.WARNING:
      return 'alert-triangle';
    default:
      return 'info';
  }
}
