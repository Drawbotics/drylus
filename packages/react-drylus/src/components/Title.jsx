import React from 'react';
import PropTypes from 'prop-types';
import sv from '@drawbotics/drylus-style-vars';
import { css, cx } from 'emotion';
import Enum from '@drawbotics/enums';

import { useResponsiveProps } from '../utils/hooks';
import { deprecateProperty } from '../utils';
import { Align } from '../enums';


const styles = {
  root: css`
    color: ${sv.colorPrimary};
    font-weight: 300;
  `,
  h1: css`
    font-size: 3.4rem;
    
    @media ${sv.screenL} {
      font-size: 2.8rem;
    }
  `,
  h2: css`
    font-size: 2.4rem;

    @media ${sv.screenL} {
      font-size: 2rem;
    }
  `,
  h3: css`
    font-size: 2rem;

    @media ${sv.screenL} {
      font-size: 1.6rem;
    }
  `,
  h4: css`
    font-size: 1.3rem;
    font-weight: 400;

    @media ${sv.screenL} {
      font-size: 1.1rem;
    }
  `,
  noMargin: css`
    margin-top: 0;
    margin-bottom: 0;
  `,
  alignRight: css`
    text-align: right;
  `,
  alignCenter: css`
    text-align: center;
  `,
};


/**
 * @deprecated and will be removed in version 6.0
 */
export const TitleAlign = deprecateProperty(new Enum(
  'CENTER',
  'LEFT',
  'RIGHT',
), 'TitleAlign', 'Align');


const Title = ({
  responsive,
  ...rest
}) => {
  const {
    children,
    size,
    noMargin,
    style,
    align,
  } = useResponsiveProps(rest, responsive);

  if (size === 1) {
    return (
      <h1
        style={style}
        className={cx(styles.root, styles.h1, {
          [styles.noMargin]: noMargin,
          [styles.alignCenter]: align === Align.CENTER,
          [styles.alignRight]: align === Align.RIGHT,
        })}>
        {children}
      </h1>
    );
  }
  else if (size === 2) {
    return (
      <h2
        style={style}
        className={cx(styles.root, styles.h2, {
          [styles.noMargin]: noMargin,
          [styles.alignCenter]: align === Align.CENTER,
          [styles.alignRight]: align === Align.RIGHT,
        })}>
        {children}
      </h2>
    );
  }
  else if (size === 3) {
    return (
      <h3
        style={style}
        className={cx(styles.root, styles.h3, {
          [styles.noMargin]: noMargin,
          [styles.alignCenter]: align === Align.CENTER,
          [styles.alignRight]: align === Align.RIGHT,
        })}>
        {children}
      </h3>
    );
  }
  else if (size === 4) {
    return (
      <h4
        style={style}
        className={cx(styles.root, styles.h4, {
          [styles.noMargin]: noMargin,
          [styles.alignCenter]: align === Align.CENTER,
          [styles.alignRight]: align === Align.RIGHT,
        })}>
        {children}
      </h4>
    );
  }
  else {
    console.warn('Unsupported title size');
    return '';
  }
};


Title.propTypes = {
  /** Text displayed by the title */
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.string,
  ]),

  /** Each number is equivalent to the h[n] in html, smaller value equals larger title */
  size: PropTypes.oneOf([1, 2, 3, 4]),

  /** Use this if you dont want the component to set margin. By default it has some top and bottom margin since it is a textual component */
  noMargin: PropTypes.bool,

  /** Used for style overrides */
  style: PropTypes.object,

  align: PropTypes.oneOf([
    Align.CENTER,
    Align.LEFT,
    Align.RIGHT,
  ]),

  /** Reponsive prop overrides */
  responsive: PropTypes.shape({
    XS: PropTypes.object,
    S: PropTypes.object,
    M: PropTypes.object,
    L: PropTypes.object,
    XL: PropTypes.object,
    HUGE: PropTypes.object,
  }),
};


Title.defaultProps = {
  size: 1,
  align: Align.LEFT, 
};


export default Title;
