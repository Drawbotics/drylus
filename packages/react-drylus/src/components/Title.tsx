import sv from '@drawbotics/drylus-style-vars';
import { css, cx } from 'emotion';
import React from 'react';

import { Align } from '../enums';
import { Responsive, Style } from '../types';
import { useResponsiveProps } from '../utils';

const styles = {
  root: css`
    color: ${sv.colorPrimary};
    font-weight: 300;
  `,
  h1: css`
    font-size: 3.4rem;

    @media ${sv.screenL} {
      font-size: 2.8rem;
    }
  `,
  h2: css`
    font-size: 2.4rem;

    @media ${sv.screenL} {
      font-size: 2rem;
    }
  `,
  h3: css`
    font-size: 2rem;

    @media ${sv.screenL} {
      font-size: 1.6rem;
    }
  `,
  h4: css`
    font-size: 1.3rem;
    font-weight: 400;

    @media ${sv.screenL} {
      font-size: 1.1rem;
    }
  `,
  noMargin: css`
    margin-top: 0;
    margin-bottom: 0;
  `,
  alignRight: css`
    text-align: right;
  `,
  alignCenter: css`
    text-align: center;
  `,
  alignJustify: css`
    text-align: justify;
  `,
};

export interface TitleProps {
  /** Text displayed by the title */
  children: React.ReactNode;

  /**
   * Each number is equivalent to the h[n] in html, smaller value equals larger title
   * @default 1
   */
  size?: 1 | 2 | 3 | 4;

  /** Use this if you dont want the component to set margin. By default it has some top and bottom margin since it is a textual component */
  noMargin?: boolean;

  /** Used for style overrides */
  style?: Style;

  /** @default Align.LEFT */
  align?: Align;

  /** Reponsive prop overrides */
  responsive?: Responsive<this>;
}

export const Title = ({ responsive, ...rest }: TitleProps) => {
  const { children, size = 1, noMargin, style, align = Align.LEFT } = useResponsiveProps<
    TitleProps
  >(rest, responsive);

  if (size === 1) {
    return (
      <h1
        style={style}
        className={cx(styles.root, styles.h1, {
          [styles.noMargin]: noMargin === true,
          [styles.alignCenter]: align === Align.CENTER,
          [styles.alignRight]: align === Align.RIGHT,
          [styles.alignJustify]: align === Align.JUSTIFY,
        })}>
        {children}
      </h1>
    );
  } else if (size === 2) {
    return (
      <h2
        style={style}
        className={cx(styles.root, styles.h2, {
          [styles.noMargin]: noMargin === true,
          [styles.alignCenter]: align === Align.CENTER,
          [styles.alignRight]: align === Align.RIGHT,
          [styles.alignJustify]: align === Align.JUSTIFY,
        })}>
        {children}
      </h2>
    );
  } else if (size === 3) {
    return (
      <h3
        style={style}
        className={cx(styles.root, styles.h3, {
          [styles.noMargin]: noMargin === true,
          [styles.alignCenter]: align === Align.CENTER,
          [styles.alignRight]: align === Align.RIGHT,
          [styles.alignJustify]: align === Align.JUSTIFY,
        })}>
        {children}
      </h3>
    );
  } else if (size === 4) {
    return (
      <h4
        style={style}
        className={cx(styles.root, styles.h4, {
          [styles.noMargin]: noMargin === true,
          [styles.alignCenter]: align === Align.CENTER,
          [styles.alignRight]: align === Align.RIGHT,
          [styles.alignJustify]: align === Align.JUSTIFY,
        })}>
        {children}
      </h4>
    );
  } else {
    console.warn('Unsupported title size');
    return null;
  }
};
