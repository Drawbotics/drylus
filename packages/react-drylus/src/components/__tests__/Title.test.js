import React from 'react';
import { create } from 'react-test-renderer';

import Title from '../Title';
import { Align } from '../../enums';


describe('Title', () => {
  describe('matches snapshot when', () => {
    it('has a h1 size', () => {
      const tree = create(
        <Title size={1}>Title content</Title>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('has a h2 size', () => {
      const tree = create(
        <Title size={2}>Title content</Title>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('has a h3 size', () => {
      const tree = create(
        <Title size={3}>Title content</Title>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('has a h4 size', () => {
      const tree = create(
        <Title size={4}>Title content</Title>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('does not have margins', () => {
      const tree = create(
        <Title noMargin>Title content</Title>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('has a node as children', () => {
      const tree = create(
        <Title size={2}>
          <span style={{ fontWeight: 500 }}>Some </span> content
        </Title>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('is aligned to the right', () => {
      const tree = create(
        <Title align={Align.RIGHT}>
         Some content
        </Title>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('is center aligned', () => {
      const tree = create(
        <Title align={Align.CENTER}>
         Some content
        </Title>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});