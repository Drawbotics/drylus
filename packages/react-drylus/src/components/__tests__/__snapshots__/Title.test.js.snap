// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title matches snapshot when does not have margins 1`] = `
.css-17thm8e-Title__root-Title__h1-Title__noMargin {
  color: #172b4e;
  font-weight: 300;
  font-size: 3.4rem;
  margin-top: 0;
  margin-bottom: 0;
}

@media only screen and (max-width:768px) {
  .css-17thm8e-Title__root-Title__h1-Title__noMargin {
    font-size: 2.8rem;
  }
}

<h1
  className="css-17thm8e-Title__root-Title__h1-Title__noMargin"
>
  Title content
</h1>
`;

exports[`Title matches snapshot when has a h1 size 1`] = `
.css-jn5qyg-Title__root-Title__h1 {
  color: #172b4e;
  font-weight: 300;
  font-size: 3.4rem;
}

@media only screen and (max-width:768px) {
  .css-jn5qyg-Title__root-Title__h1 {
    font-size: 2.8rem;
  }
}

<h1
  className="css-jn5qyg-Title__root-Title__h1"
>
  Title content
</h1>
`;

exports[`Title matches snapshot when has a h2 size 1`] = `
.css-1d267us-Title__root-Title__h2 {
  color: #172b4e;
  font-weight: 300;
  font-size: 2.4rem;
}

@media only screen and (max-width:768px) {
  .css-1d267us-Title__root-Title__h2 {
    font-size: 2rem;
  }
}

<h2
  className="css-1d267us-Title__root-Title__h2"
>
  Title content
</h2>
`;

exports[`Title matches snapshot when has a h3 size 1`] = `
.css-eqynv4-Title__root-Title__h3 {
  color: #172b4e;
  font-weight: 300;
  font-size: 2rem;
}

@media only screen and (max-width:768px) {
  .css-eqynv4-Title__root-Title__h3 {
    font-size: 1.6rem;
  }
}

<h3
  className="css-eqynv4-Title__root-Title__h3"
>
  Title content
</h3>
`;

exports[`Title matches snapshot when has a h4 size 1`] = `
.css-ucdkco-Title__root-Title__h4 {
  color: #172b4e;
  font-weight: 300;
  font-size: 1.3rem;
  font-weight: 400;
}

@media only screen and (max-width:768px) {
  .css-ucdkco-Title__root-Title__h4 {
    font-size: 1.1rem;
  }
}

<h4
  className="css-ucdkco-Title__root-Title__h4"
>
  Title content
</h4>
`;

exports[`Title matches snapshot when has a node as children 1`] = `
.css-1d267us-Title__root-Title__h2 {
  color: #172b4e;
  font-weight: 300;
  font-size: 2.4rem;
}

@media only screen and (max-width:768px) {
  .css-1d267us-Title__root-Title__h2 {
    font-size: 2rem;
  }
}

<h2
  className="css-1d267us-Title__root-Title__h2"
>
  <span
    style={
      Object {
        "fontWeight": 500,
      }
    }
  >
    Some 
  </span>
   content
</h2>
`;
